Modules/_ctypes/_ctypes.c

```diff
@@ -1750,7 +1750,11 @@ class _ctypes.c_void_p "PyObject *" "clinic_state_sub()->PyCSimpleType_Type"
 [clinic start generated code]*/
 /*[clinic end generated code: output=da39a3ee5e6b4b0d input=dd4d9646c56f43a9]*/
 
+#ifdef __STDC_IEC_559_COMPLEX__
```

Comment:
You should also check the presence of `__STDC_NO_COMPLEX__` (see https://en.cppreference.com/w/c/numeric/complex).

Reply:
Both could be defined only by some broken compiler.  Do you know examples?

Reply:
> An implementation that defines `__STDC_NO_COMPLEX__` shall not define `__STDC_IEC_559_COMPLEX__`

You're right, it's my bad!

Reply:
On another hand, probably this test too naive and we should check instead e.g. for ``defined(_Complex_I) && defined(complex) && defined(I)`` (as GSL does).

For C11 and C2X - imaginary types are mandatory.  So, definition of the ``__STDC_IEC_559_COMPLEX__`` macro - actually is a bug without imaginary types.  (See e.g. https://github.com/llvm/llvm-project/issues/60269.)

Reply:
@picnixz, now I test for _Complex type in configure.  Probably this better fits your suggestion.  I test pass - HAVE_C_COMPLEX defined.  Workarounds for buggy implementations moved to Modules/_complex.h.